### PR TITLE
Issue #12364: Modification on how to run checkstyle using CLI

### PR DESCRIPTION
**Pull request by NitishKumar06**
Fix: #12364

### CODE REVIEW

**Comment by NitishKumar06** [comment]
GitHub generate website

File: src/xdocs/cmdline.xml.vm [hunk 1]
```diff
@@ -169,27 +169,33 @@ java -D&lt;property&gt;=&lt;value&gt;  \
...
+        <p>
+        1. Download latest
         <a href="https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${projectVersion}/checkstyle-${projectVersion}-all.jar">
-        checkstyle-${projectVersion}-all.jar</a>.
-        An example of run would be:
-      </p>
+           checkstyle-${projectVersion}-all.jar</a>.
+        </p>
+        <p>
+        2. Create a configuration file required to run Checkstyle.
```
Comment (romani) [inline src/xdocs/cmdline.xml.vm:183]: This example does not need config file creation as it uses already prepared embedded config.

Attention to `/` before name of file. It signals to load it from classpath/jar too.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/cmdline.xml.vm (hunk 1 above)
Comment (NitishKumar06) [reply src/xdocs/cmdline.xml.vm:183]: Sounds good. I may remove the 2. point I've mentioned. Any more suggestions ?

@rnveach Any idea on why all the jobs are failing? The site has been built successfully though.

Comment (rnveach) [reply src/xdocs/cmdline.xml.vm:183]: Look at CI. `xml-maven-plugin` is failing from your changes.

Comment (romani) [reply src/xdocs/cmdline.xml.vm:183]: Please remove

Comment (NitishKumar06) [reply src/xdocs/cmdline.xml.vm]: @romani The changes have been applied now! Kindly take a look.

**Comment by NitishKumar06** [comment]
github generate website

**Comment by NitishKumar06** [comment]
GitHub generate website

File: src/xdocs/cmdline.xml.vm [hunk 2]
```diff
@@ -169,27 +169,32 @@ java -D&lt;property&gt;=&lt;value&gt;  \
...
+      <p>
+        3. Run Checkstyle using either of the commands:
       </p>
       <div class="wrap-content">
         <source>
           java -jar checkstyle-${projectVersion}-all.jar -c /sun_checks.xml MyClass.java
           java -jar checkstyle-${projectVersion}-all.jar -c /google_checks.xml MyClass.java
         </source>
       </div>
-      <p>
```
Comment (romani) [inline src/xdocs/cmdline.xml.vm:183]: Keep recommendation after first execution, no need to overwhelm user with details at the beginning

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@NitishKumar06, please rebase on our latest master, you PR is old.

**Comment by NitishKumar06** [comment]
@romani DONE!

**Comment by romani** [comment]
Not done. No code changes in GitHub.

**Comment by NitishKumar06** [comment]
@romani done!

File: src/xdocs/cmdline.xml.vm [hunk 3]
```diff
@@ -168,27 +168,28 @@ java -D&lt;property&gt;=&lt;value&gt;  \
...
+      </p>
+      <p>
+        1. Download latest
         <a href="https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${projectVersion}/checkstyle-${projectVersion}-all.jar">
-        checkstyle-${projectVersion}-all.jar</a>.
-        An example of run would be:
+           checkstyle-${projectVersion}-all.jar</a>.
+           It is recommended to use configuration files that are embedded in jar files, but latest
+           configuration files are there:<a href="https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/sun_checks.xml">
+           sun_checks.xml</a> <a href="https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml">google_checks.xml</a>
```
Comment (romani) [inline src/xdocs/cmdline.xml.vm:182]: Please format links as it was before to take less horizontal space

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Review by romani — APPROVED** [review]

File: src/xdocs/cmdline.xml.vm [hunk 4]
```diff
@@ -168,27 +168,33 @@ java -D&lt;property&gt;=&lt;value&gt;  \
...
+           configuration files can be found here:
+        <a href=
+           "https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/sun_checks.xml">
+             sun_checks.xml</a>
+        <a href=
+           "https://raw.githubusercontent.com/checkstyle/checkstyle/master/src/main/resources/google_checks.xml">
+             google_checks.xml</a>
+      </p>
+      <p>
+        3. Run Checkstyle using either of the commands:
```
Comment (rnveach) [inline src/xdocs/cmdline.xml.vm:190]: There is no step 2, we go from step 1 to step 3.

File: src/xdocs/cmdline.xml.vm [hunk 5]
```diff
@@ -168,27 +168,33 @@ java -D&lt;property&gt;=&lt;value&gt;  \
...
+      <p>
+        Follow the steps mentioned below to run Checkstyle locally:
+      </p>
+      <p>
+        1. Download latest
         <a href="https://github.com/checkstyle/checkstyle/releases/download/checkstyle-${projectVersion}/checkstyle-${projectVersion}-all.jar">
-        checkstyle-${projectVersion}-all.jar</a>.
-        An example of run would be:
+           checkstyle-${projectVersion}-all.jar</a>.
+           It is recommended to use configuration files that are embedded in jar files, but latest
```
Comment (rnveach) [inline src/xdocs/cmdline.xml.vm:180]: This isn't even a step 2, it is more a note which I think still makes sense being after the commands.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/issues/12364#issuecomment-1304505152
> ...add a bit more detail about extracting configs from jar file...

I am not seeing any additions of this either. It was just moving existing documentation which I felt the original position might be better.

**Comment by NitishKumar06** [comment]
> 

The main aim was to make the documentation clear. Hence It was added in the form of a number list.

**Comment by nrmancuso** [comment]
Github, generate site

**Comment by nrmancuso** [comment]
@NitishKumar06 read your changes carefully, and reread comment at https://github.com/checkstyle/checkstyle/pull/12378#discussion_r1027324816

Additionally, never remove maintainers from review list.

**Comment by NitishKumar06** [comment]
@nrmancuso I've applied minor changes in the latest PR.

Let me know what more content you should be added.

**Comment by nrmancuso** [comment]
Github, generate site

**Comment by nrmancuso** [comment]
Before this PR (https://checkstyle.sourceforge.io/cmdline.html#Download_and_Run):

image


After this PR (https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9a6a478_2022151000/cmdline.html#Download_and_Run):

image

File: src/xdocs/cmdline.xml.vm (hunk 5 above)
Comment (nrmancuso) [reply src/xdocs/cmdline.xml.vm:180]: See comment at https://github.com/checkstyle/checkstyle/pull/12378#issuecomment-1328271903, I have to agree here. It looks like we just enumerated the information that was already present, and moved some elements around.


I had hoped for some grammatical improvements and a bit of explanation about the `It is recommended to use configuration files that are embedded in jar files` part. It took me a while to realize that the user wasn't supposed to hack the jar file and extract the configs themselves.

**Comment by nrmancuso** [comment]
We have lost contact with author, I am closing this PR.

**Pull request closed without merging**